spring-shell2-core/src/main/java/org/springframework/shell2/standard/StandardParameterResolver.java

```diff
@@ -98,7 +98,7 @@ public void setValueProviders(Collection<ValueProvider> valueProviders) {
 
 	@Override
 	public boolean supports(MethodParameter parameter) {
-		return true;
+		return parameter.getMethodAnnotation(ShellMethod.class) != null;
```

Comment:
Ah, sorry, did not read right.
So for now, this is indeed fine and needed (once we derive `help` from other sources (see #13) we may have to revisit, but totally fine by now)